core/tests/test_popular_search_terms.py

```
@@ -0,0 +1,833 @@
+
+    return PopularSearchTermAdmin(PopularSearchTerm, AdminSite())
+
+
+def _admin_request(method="get", path="/admin/core/popularsearchterm/", **post):
+    """RequestFactory request with the messages framework wired up."""
+    from django.contrib.messages.storage.fallback import FallbackStorage
+    from django.test import RequestFactory
+
+    rf = RequestFactory()
+    request = getattr(rf, method)(path, post) if method == "post" else rf.get(path)
+    setattr(request, "session", {})
```

Ruff (0.15.17)

[warning] Do not call `setattr` with a constant attribute value. It is not any safer than normal property access.

Replace `setattr` with assignment

(B010)